lib/ansible/modules/windows/setup.ps1

```diff
@@ -349,7 +349,7 @@ if($gather_subset.Contains('processor')) {
     }
 
     $cpu_list = @( )
-    for ($i=1; $i -le ($win32_cpu.NumberOfLogicalProcessors / $win32_cs.NumberOfProcessors); $i++) {
+    for ($i=1; $i -le ($win32_cpu.NumberOfLogicalProcessors * $win32_cs.NumberOfProcessors); $i++) {
```

Comment:
In my testing this could just be `for ($i=i; $i -lt $win32_cs.NumberOfLogicalProcessors; $i++) {`

This was we don't need to do the calculation as `Win32_ComputerSystem` has already done it for us.